src/games/gameLogic.ts

```diff
@@ -1,64 +1,53 @@
-const board1 = [
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0']
-]
+import { ValidatorConstraint, ValidatorConstraintInterface } from 'class-validator'
+import { Board, emptyBoard, Row} from './entities';
 
-const board2 = [
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0'],
-  ['0', '0', '0', '0', '0', '0', '0', '0', '0', '0']
-]
+@ValidatorConstraint()
+export class IsBoard implements ValidatorConstraintInterface {
+
+  validate(board: Board) {
+    return board.length === 10 &&
+      board.every(row =>
+        row.length === 10 &&
+        row.every(value => typeof(value)==='string')
+      )
+  }
+}
```

Comment:
maybe I should have added the previous comment here

Reply:
now the upper half of the code is missing hahaha

Reply:
No worry it's fine too :smiley: